strawberry/starlite/controller.py

```diff
@@ -0,0 +1,399 @@
+                context.response = response
+            elif isinstance(context, dict):
+                context["response"] = response
+            try:
+                result = await self.execute(
+                    request_data.query,
+                    variables=request_data.variables,
+                    context=context,
+                    operation_name=request_data.operation_name,
+                    root_value=root_value,
+                    allowed_operation_types=allowed_operation_types,
+                )
+            except InvalidOperationTypeError as e:
+                return Response(
+                    e.as_http_error_reason(method),
+                    status_code=HTTP_400_BAD_REQUEST,
+                    media_type=MediaType.TEXT,
+                )
+
+            response_data = await self.process_result(result)
+
+            actual_response: Response[GraphQLHTTPResponse] = Response(
+                response_data, status_code=HTTP_200_OK, media_type=MediaType.JSON
+            )
+
+            return self._merge_responses(response, actual_response)
+
+        def should_render_graphiql(self, request: Request) -> bool:
+            if not self._graphiql:
+                return False
+            return any(
+                supported_header in request.headers.get("accept", "")
+                for supported_header in ("text/html", "*/*")
+            )
+
+        def get_graphiql_response(self) -> Response[str]:
+            html = get_graphiql_html()
+            return Response(html, media_type=MediaType.HTML)
+
+        @staticmethod
+        def _merge_responses(
+            response: Response, actual_response: Response
+        ) -> Response[Union[GraphQLResource, str]]:
+            actual_response.headers.update(response.headers)
+            actual_response.cookies.extend(response.cookies)
+            actual_response.background = response.background
+            if response.status_code:
+                actual_response.status_code = response.status_code
+
+            return actual_response
+
+        @get(
+            responses={
+                HTTP_404_NOT_FOUND: ResponseSpec(
+                    description="Not found if GraphiQL is not enabled.",
+                    model=EmptyResponseModel,
+                    media_type=MediaType.TEXT,
+                ),
+                HTTP_400_BAD_REQUEST: ResponseSpec(
+                    description="Unable to parse request body as JSON",
+                    model=EmptyResponseModel,
+                    media_type=MediaType.TEXT,
+                ),
+            },
+        )
+        async def handle_http_get(
+            self,
+            request: Request,
+            context: CustomContext,
+            root_value: Any,
+        ) -> Response[Union[GraphQLResource, str]]:
+            actual_response: Response
+
+            if request.query_params:
+                try:
+                    query_data = parse_query_params(
```

Comment:
`request.query_params` is already a parsed dict. It will not raise a json error in any case since we don't use json there.